packages/cli-kit/src/public/node/http.ts

```diff
@@ -28,45 +155,58 @@
  * they are consistent with the Web API so if we drop node-fetch in the future
  * it won't require changes from the callers.
  *
+ * The CLI's fetch function supports special behaviours, like automatic retries. These are disabled by default through
+ * this function.
+ *
  * @param url - This defines the resource that you wish to fetch.
  * @param init - An object containing any custom settings that you want to apply to the request.
+ * @param preferredBehaviour - A request behaviour object that overrides the default behaviour.
  * @returns A promise that resolves with the response.
  */
-export async function fetch(url: RequestInfo, init?: RequestInit): Promise<Response> {
-  return runWithTimer('cmd_all_timing_network_ms')(() =>
-    simpleRequestWithDebugLog({url: url.toString(), request: () => nodeFetch(url, init)}),
-  )
+export async function fetch(
+  url: RequestInfo,
+  init?: RequestInit,
+  preferredBehaviour?: RequestModeInput,
+): Promise<Response> {
+  const options = {
+    url,
+    init,
+    logRequest: false,
+    useHttpsAgent: false,
+    // all special behaviours are disabled by default
+    behaviour: preferredBehaviour ? requestMode(preferredBehaviour) : requestMode('non-blocking'),
+  } as const
+
+  return innerFetch(options)
 }
 
 /**
  * A fetch function to use with Shopify services. The function ensures the right
  * TLS configuragion is used based on the environment in which the service is running
- * (e.g. Spin).
+ * (e.g. Spin). NB: headers/auth are the responsibility of the caller.
+ *
+ * By default, the CLI's fetch function's special behaviours, like automatic retries, are enabled.
```

Comment:
Why is it enabled here but disabled in the `fetch` above by default?
`shopifyFetch` is only used for identity i think no?

Reply:
I see, the only difference is that shopifyFetch will inject some custom headers. The reasoning makes sense but the truth is that we are using `fetch` also for some shopify APIs :( 
 
Requests using `fetch` that are actually connecting to shopify services:
- `admin.ts` to run queries to the Admin API.
- `storefront-session.ts` to connect to the storefront API
- `utils.ts` to fetch the app logs
- I'm almost sure that Hydrogen is also using it for internal stuff (because they don't use `shopifyFetch`)

So I think we either should:
- Migrate all these to use `shopifyFetch`
- Enable the `default` mode for `fetch` (and maybe deprecate `shopifyFetch`?)

wdyt?